lib/core/versions/latest/ValueTimeLockVerifier.ts

```diff
@@ -0,0 +1,77 @@
+import ErrorCode from './ErrorCode';
+import ProtocolParameters from './ProtocolParameters';
+import SidetreeError from '../../../common/SidetreeError';
+import ValueTimeLockModel from '../../../common/models/ValueTimeLockModel';
+
+/**
+ * Encapsulates the functionality to compute and verify the value time lock amounts.
+ */
+export default class ValueTimeLockVerifier {
+
+  /**
+   * Calculates the maximum number of operations allowed to be written for the given lock information. If
+   * there is no lock then it returns the number of operations which do not require a lock.
+   *
+   * @param valueTimeLock The lock object if exists
+   * @param normalizedFee The normalized fee for the current block
+   */
+  public static calculateMaxNumberOfOpsAllowed (valueTimeLock: ValueTimeLockModel | undefined, normalizedFee: number) {
+
+    if (valueTimeLock === undefined) {
+      return ProtocolParameters.maxNumberOfOpsForNoValueTimeLock;

```

Comment:
In the spirit of not introducing further parameters unless we must, do we want to derive `maxNumberOfOpsForNoValueTimeLock` from `normalizedFeeToPerOperationFeeMultiplier`?

If we change `normalizedFeeToPerOperationFeeMultiplier` to `normalizedFeeToPerOperationFeeDivisor`, then we can just use the value as is.

Reply:
Keeping the two values separate to keep the max-free-ops and normalized-fee separate.